lib/fable/events.ex

```diff
@@ -105,43 +205,73 @@ defmodule Fable.Events do
     |> Enum.map(&Fable.Event.parse_data(config.repo, &1))
   end
 
-  def emit!(config, schema, event, opts \\ []) do
-    case emit(config, schema, event, opts) do
-      {:ok, schema} ->
-        schema
+  @doc false
+  @spec emit(Fable.Config.t(), aggregate_source, event_or_events, Keyword.t()) :: transation_fun
+  def unconditionally_emit(config, schema_or_queryable, event_or_events, opts \\ []) do
+    emit(config, schema_or_queryable, fn _, _, _ -> event_or_events end, opts)
+  end
+
+  @doc false
+  @spec emit(Fable.Config.t(), aggregate_source, emit_callback, Ecto.Multi.changes(), Keyword.t()) ::
+          transation_fun
 
-      error ->
-        raise error
+  def emit(config, schema_or_queryable, fun, changes \\ %{}, opts)
+
+  def emit(%{repo: repo} = config, %Ecto.Query{} = queryable, fun, changes, opts)
```

Comment:
Per the other comment on queryables, I can't see a reason why we should support queryables.